Assets/LeapMotion/Modules/InteractionEngine/Testing/Scripts/Tests/InteractionButtonTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿/******************************************************************************
+ * Copyright (C) Leap Motion, Inc. 2011-2017.                                 *
+ * Leap Motion proprietary and  confidential.                                 *
+ *                                                                            *
+ * Use subject to the terms of the Leap Motion SDK Agreement available at     *
+ * https://developer.leapmotion.com/sdk_agreement, or another agreement       *
+ * between Leap Motion and you, your company or other organization.           *
+ ******************************************************************************/
+
+#if LEAP_TESTS
+
+using NUnit.Framework;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Leap.Unity.Interaction.Tests {
+
+  public class InteractionButtonTests : InteractionEngineTestBase {
+    
+    #region Primary Hover Lock
+
+    [UnityTest]
+    public IEnumerator CanDestroyButtonWhenPressed() {
+      yield return wait(beginningTestWait);
+
+      InitTest(PRESS_BUTTON_RIG, DEFAULT_STAGE);
+
+      // Wait before starting the test.
+      yield return wait(aWhile);
+
+      // Play the grasping animation.
+      recording.Play();
+
+      // Schedule the deletion of the button when the button is pressed.
+      bool pressed = false;
+      button.OnPress += () => {
+        pressed = true;
+
+        GameObject.Destroy(button);
+      };
+
+      // Wait until the button is pressed.
+      int framesWaited = 0;
+      while (!pressed && framesWaited < WAIT_FOR_INTERACTION_FRAME_LIMIT) {
+        yield return null;
+        framesWaited++;
+      }
+      Assert.That(framesWaited != WAIT_FOR_INTERACTION_FRAME_LIMIT);
```

Comment:
Would be good to add in the optional string params to each of these asserts, that way if they fail sometime later it is easier to know what exactly is going wrong.  Currently, it will say something like 'expected true but was false', would be nice if it said something like 'recording failed to press button'.